cartography/intel/aws/ec2/instances.py

```diff
@@ -21,410 +29,343 @@
 def get_ec2_instances(boto3_session: boto3.session.Session, region: str) -> List[Dict]:
     client = boto3_session.client('ec2', region_name=region, config=get_botocore_config())
     paginator = client.get_paginator('describe_instances')
-    reservations: List[Dict] = []
+    reservations: List[Dict[str, Any]] = []
     for page in paginator.paginate():
         reservations.extend(page['Reservations'])
     return reservations
 
 
-def _load_ec2_instance_net_if_tx(
-        tx: neo4j.Transaction,
-        instance_data: Dict[str, Any],
-        update_tag: int,
-) -> None:
-    query = """
-    MATCH (instance:EC2Instance{instanceid: $InstanceId})
-    UNWIND $Interfaces as interface
-        MERGE (nic:NetworkInterface{id: interface.NetworkInterfaceId})
-        ON CREATE SET nic.firstseen = timestamp()
-        SET nic.status = interface.Status,
-        nic.mac_address = interface.MacAddress,
-        nic.description = interface.Description,
-        nic.private_dns_name = interface.PrivateDnsName,
-        nic.private_ip_address = interface.PrivateIpAddress,
-        nic.lastupdated = $update_tag
-
-        MERGE (instance)-[r:NETWORK_INTERFACE]->(nic)
-        ON CREATE SET r.firstseen = timestamp()
-        SET r.lastupdated = $update_tag
-
-        WITH nic, interface
-        WHERE interface.SubnetId IS NOT NULL
-        MERGE (subnet:EC2Subnet{subnetid: interface.SubnetId})
-        ON CREATE SET subnet.firstseen = timestamp()
-        SET subnet.lastupdated = $update_tag
-
-        MERGE (nic)-[r:PART_OF_SUBNET]->(subnet)
-        ON CREATE SET r.firstseen = timestamp()
-        SET r.lastupdated = $update_tag
-
-        WITH nic, interface
-        UNWIND interface.Groups as group
-            MATCH (ec2group:EC2SecurityGroup{groupid: group.GroupId})
-            MERGE (nic)-[r:MEMBER_OF_EC2_SECURITY_GROUP]->(ec2group)
-            ON CREATE SET r.firstseen = timestamp()
-            SET r.lastupdated = $update_tag
-    """
-    tx.run(
-        query,
-        Interfaces=instance_data['NetworkInterfaces'],
-        InstanceId=instance_data['InstanceId'],
-        update_tag=update_tag,
-    )
+def transform_ec2_instances(
```

Comment:
@ramonpetgrave64 - what you think about this big function, should I split it?

Reply:
I'd rather that return such a big tuple. Maybe convert it to a namedtuple, so we don't have to be so careful about the ordering?

It's a big one, but if we break it into many functions, we would be repeating a bit of code to  dig deep into the objects. I'm okay with it either way.